demos/notebooks/demo_pipeline_cnmfE.ipynb

```diff
@@ -69,16 +82,85 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "fnames = ['data_endoscope.tif']  # filename to be processed\n",
-    "fnames = [download_demo(fnames[0])]"
+    "# set up logging\n",
+    "logging.basicConfig(format=\"{asctime} - {levelname} - [{filename} {funcName}() {lineno}] - pid {process} - {message}\",\n",
+    "                    filename=None, \n",
+    "                    level=logging.WARNING, style=\"{\") #logging level can be DEBUG, INFO, WARNING, ERROR, CRITICAL\n",
+    "\n",
+    "# set env variables \n",
+    "os.environ[\"MKL_NUM_THREADS\"] = \"1\"\n",
+    "os.environ[\"OPENBLAS_NUM_THREADS\"] = \"1\"\n",
+    "os.environ[\"VECLIB_MAXIMUM_THREADS\"] = \"1\""
    ]
   },
   {
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "### Setup a cluster\n",
-    "To enable parallel processing a (local) cluster needs to be set up. This is done with a cell below. The variable `backend` determines the type of cluster used. The default value `'multiprocessing'` uses the multiprocessing package. The `ipyparallel` option is also available. More information on these choices can be found [here](https://github.com/flatironinstitute/CaImAn/blob/master/CLUSTER.md). The resulting variable `dview` expresses the cluster option. If you use `dview=dview` in the downstream analysis then parallel processing will be used. If you use `dview=None` then no parallel processing will be employed."
+    "# Select file(s) to be processed\n",
+    "Here, we analyze the data in `data_endoscope.tif`. The `download_demo` function will download the  file for you and return the complete path to the file which will be stored in your `caiman_data` directory. If you adapt this demo for your data make sure to pass the complete path to your file. \n",
```

Comment:
Option: reference the function name as download_demo()

(you don't need to do this, just might be worth thinking about it)